RealmWeaver/ModuleWeaver.cs

```diff
@@ -303,29 +301,31 @@ private void WeaveType(TypeDefinition type, Dictionary<string, Tuple<MethodRefer
 
             ReplaceGetter(prop, columnName, methodTable[typeId].Item1);
             ReplaceSetter(prop, backingField, columnName, methodTable[typeId].Item2, typeImplementsPropertyChanged, propChangedEventDefinition, propChangedFieldDefinition);
-        }
-//                else if (prop.PropertyType.Name == "IList`1" && prop.PropertyType.Namespace == "System.Collections.Generic")
-        else if (prop.PropertyType.Name == "RealmList`1" && prop.PropertyType.Namespace == "Realms")
+        } 
+
+        // treat IList and RealmList identically
+        else if ((prop.PropertyType.Name == "IList`1" && prop.PropertyType.Namespace == "System.Collections.Generic")  ||
+            (prop.PropertyType.Name == "RealmList`1" && prop.PropertyType.Namespace == "Realms"))
         {
-            // RealmList allows people to declare lists only of _realmObject due to the class definition
-            if (!prop.IsAutomatic())
-            {
-                LogErrorPoint(
-                    $"{type.Name}.{columnName} is not an automatic property but its type is a RealmList which normally indicates a relationship",
-                    sequencePoint);
+            var elementType = ((GenericInstanceType)prop.PropertyType).GenericArguments.Single();
+            // IList or RealmList allows people to declare lists only of _realmObject due to the class definition
+            if (!elementType.Resolve().BaseType.IsSameAs(_realmObject)) {
+                LogWarningPoint(
```

Comment:
I think we should look into creating a Roslyn Code Analyzer plugin which will be emitting warnings instead of the weaver. Because it ties into the IDEs people will be able to silence those. It's nice for us to be verbose about what we don't persist in the weaver with warnings and all, but right now there's no way to turn off these warnings so more experienced developers are going to be annoyed if we issue too many warnings from the weaver.